src/main/java/com/github/matidominati/medicalclinic/personInfo/PatientController.java

```diff
@@ -0,0 +1,55 @@
+package com.github.matidominati.medicalclinic.personInfo;
+
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/patients")
+public class PatientController {
+    private List<Patient> patients = new ArrayList<>();
+
+    @GetMapping
+    public List<Patient> getAllPatients() {
+        return patients;
+    }
+
+    @GetMapping("/{email}")
+    public Patient getPerson(@PathVariable String email) {
+        return patients.stream()
+                .filter(patient -> patient.getEmail().equals(email))
+                .findFirst()
+                .orElseThrow(() -> new RuntimeException("Pacjent nie istnieje w bazie"));
+
+    }
+
+    @PostMapping
+    public void addPatient(@RequestBody Patient patient) {
+        if (!patients.contains(patient)) {
+            patients.add(patient);
+        } else throw new RuntimeException("Pacjent już istnieje w bazie");
```

Comment:
Raczej nie uzywamy konstrukcji if badz else bez {}

Zakladasz ze email jest final czyli nie mozna go zmienic po dodaniu pacjenta aczkolwiek dopuszczasz sytuacje utworzenia pacjenta z emailem == null co wplywa na to, ze taki jeden pacjent juz ci wywali funkcjonalnosc pobierania pacjenta po emailu. -> NPE (Null Pointer Exception)